x-pack/plugins/encrypted_saved_objects/server/config.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import crypto from 'crypto';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { schema, TypeOf } from '@kbn/config-schema';
+import { PluginInitializerContext } from 'src/core/server';
+
+export type ConfigType = ReturnType<typeof createConfig$> extends Observable<infer P>
+  ? P
+  : ReturnType<typeof createConfig$>;
+
+export const ConfigSchema = schema.object({
+  enabled: schema.boolean({ defaultValue: true }),
+  encryptionKey: schema.conditional(
+    schema.contextRef('dist'),
```

Comment:
>  Are those the only two contexts that you're aware of?

Core passes a bunch various context variables [1](https://github.com/elastic/kibana/blob/master/src/core/server/config/config_service.ts#L145-L149) and [2](https://github.com/elastic/kibana/blob/master/src/core/server/config/env.ts#L124-L130).

> Does it matter which context you test for in different scenarios? I can surmise that dist is the built distribution, and dev is presumably dev mode, but I just wanted to make sure I'm not missing any nuances here.

Never thought about this to be honest, `$dist` that we always had in the past (e.g. in legacy security and reporting plugins) sounds more appropriate for things like this since it's something we distribute to others, but :man_shrugging:

Reply:
Ah, I see we have `prod` as a context variable. I wonder if that would be more appropriate? Or would that complicate our functional test setup? It seems like we would always want `prod` and `dist` to behave the same way in terms of configuration, but maybe not? I'm splitting hairs, I can leave this alone :)

Reply:
Hmm, these are all valid questions, let's ask Operations team! 

Hey @elastic/kibana-operations can you please explain the conceptual difference between `prod` and `dist` Kibana "context flags" and when we should use one vs another? I see we historically used `dist` for various cases (auto set encryption keys or disable advanced CORS settings if not "dist").